InfernalRobotics/InfernalRobotics/API/IRWrapper.cs

```diff
@@ -104,66 +101,74 @@ public static Boolean InitWrapper()
+                else
+                    ActualServoController = propertyInfo.GetValue(null, null);
             }
             catch (Exception e)
             {
                 LogFormatted("No Instance found, " + e.Message);
             }
 
-            if (actualServoController == null)
+            if (ActualServoController == null)
             {
                 LogFormatted("Failed grabbing Instance");
                 return false;
             }
 
             LogFormatted("Got Instance, Creating Wrapper Objects");
-            IRController = new IRAPI(actualServoController);
+            IRController = new InfernalRoboticsAPI(ActualServoController);
             isWrapped = true;
             return true;
         }
 
-        public class IRAPI
+        #region Private Implementation
+        private class InfernalRoboticsAPI : IRAPI
         {
-            internal IRAPI(Object IRServoController)
-            {
-                actualServoController = IRServoController;
+            private PropertyInfo apiReady;
+            private object actualServoGroups;
 
-                LogFormatted("Getting APIReady Object");
-                APIReadyProperty = IRServoControllerType.GetProperty("APIReady", BindingFlags.Public | BindingFlags.Static);
-                LogFormatted("Success: " + (APIReadyProperty != null).ToString());
+            public InfernalRoboticsAPI(object irServoController)
+            {
+                DetermineReady();
+                BuildServoGroups(irServoController);
+            }
 
+            private void BuildServoGroups(object irServoController)
+            {
                 LogFormatted("Getting ServoGroups Object");
-                ServoGroupsField = IRServoControllerType.GetField("ServoGroups");
-                if (ServoGroupsField == null)
+                var servoGroupsField = IRServoControllerType.GetField("ServoGroups");
+                if (servoGroupsField == null)
                     LogFormatted("Failed Getting ServoGroups fieldinfo");
-                actualServoGroups = ServoGroupsField.GetValue(actualServoController);
-                LogFormatted("Success: " + (actualServoGroups != null).ToString());
-                
+                else
+                {
+                    actualServoGroups = servoGroupsField.GetValue(irServoController);
+                    LogFormatted("Success: " + (actualServoGroups != null));
+                }
             }
 
-            private Object actualServoController;
+            private void DetermineReady()
+            {
+                LogFormatted("Getting APIReady Object");
+                apiReady = IRServoControllerType.GetProperty("APIReady", BindingFlags.Public | BindingFlags.Static);
+                LogFormatted("Success: " + (apiReady != null));
+            }
 
-            private PropertyInfo APIReadyProperty;
-            public Boolean APIReady
+            public bool Ready
```

Comment:
Ah, this is not part of the API actually, it is just the ServoController's flag meaning that it is ready to accept API commands. It can be left private as is only used from within IRWrapper